straxen/corrections_services.py

```diff
@@ -98,34 +98,65 @@ def _get_correction(self, run_id, correction, global_version):
         Smart logic to get correction from DB
         :param run_id: run id from runDB
         :param correction: correction's name, key word (str type)
-        :param global_version: global version (str type)
+        :param global_version: global version (str type) it can be also a local version
         :return: correction value(s)
         """
         when = self.get_start_time(run_id)
-        df_global = self.interface.read('global' if self.is_nt else 'global_xenon1t')
 
-        try:
-            values = []
-            for it_correction, version in df_global.iloc[-1][global_version].items():
-                if correction in it_correction:
-                    df = self.interface.read(it_correction)
-                    if global_version in ('ONLINE', 'xenonnt_temporary_five_pmts'):
-                        # We don't want to have different versions based
-                        # on when something was processed therefore
-                        # don't interpolate but forward fill.
-                        df = self.interface.interpolate(df, when, how='fill')
+        if 'local' in global_version:
+            try:
+                values = []
+                # hack to workaround "local version" for pmt gains
+                # because every pmt is its own dataframe...of course
+                version = global_version[6:] # remove local
+                if correction in {'pmt', 'n_veto', 'mu_veto'}:
+                    # get lists of pmts 
+                    df_global = self.interface.read('global' if self.is_nt else 'global_xenon1t')
+                    gains = df_global['ONLINE'][0]
+                    pmts = list(gains.keys())
+                    for it_correction in pmts: # loop over all PMTs
+                        if correction in it_correction:
+                            df = self.interface.read(it_correction)
+                            df = self.interface.interpolate(df, when)
+                            values.append(df.loc[df.index == when, version].values[0])
+                else:
+                    df = self.interface.read(correction)
```

Comment:
Hej guys, I started looking into the latest changes related to the local version. I think in general the code can be simpler and less nested. Before I make some detailed suggestion I have a question though. I am a bit confused about the PMT part as we also have the function `get_pmt_gains`. What is the difference, what are the plans here? 
Further, I am wondering if a general change on how we query the corrections would be beneficial. Currently, we do local_version -> global_version, query with global version. But I think this is unnecessarily complicated. In the future we will always query via the local_versions (which feels much more natural to me). Should not we reverse the logical structure:
* local_version -> query local_version
* global_version -> convert to local versions -> query with local_versions.
In that case we can move the "convert to local versions" part in a separate function making the code much cleaner and more readable. Are there any technical constraints preventing us from doing this?

Reply:
Hi Daniel,

Thanks for the useful comments, I'm pretty sure that the code can be improved, regarding your comment about local and global I have a few comments.
We always query local version at the end, global is just a map to a set of local versions, the way I implemented allows the use of either use global version or local version, I think we want to have both options. Specially for bookkeeping with global version we(production) don't have to worry about every single local version.
If we only query local version then we will have to build that map to get the set of global versions, so I don't see the advantage of doing that. 
PMTs are special case since every single PMT is its own dataframe wich complicates things

Reply:
I have included a few examples regarding the uses of local and global at the end of the notebook [here](https://github.com/XENONnT/corrections/blob/master/CMT/examples/CMT_user_guide.ipynb) if you want to check it